### ISSUE TITLE
GradDrop renormalization

### ISSUE THREAD

**Issue by thomsonn** [tensorflow/lingvo#254]
@jngiam Is this correct? I would expect the norms in the quotient to be the other way around.
https://github.com/tensorflow/lingvo/blob/cbceb3add8932554cb7a0aaa1a823d58fbd2b59e/lingvo/core/graddrop.py#L184-L185

Thanks

**Comment by jngiam** [comment]
Thanks for the question; what do you mean to be the other way around?  (what expression did you expect instead)

**Comment by thomsonn** [comment]
Thanks for replying! In the paper it says:
> For many of our experiments, we renormalize the final gradients so that ||∇||₂ remains constant
> throughout the GradDrop process. Although not practically required, this ensures that GradDrop does
> not alter the global learning rate and thus observed benefits result purely from GradDrop masking.

I'd expect that to renormalize in a way which would keep the gradient norm constant between the start and end of the `_Gradient` call, the code should be:
```python
 return transformed_grad * original_grad_norm / ( 
     transformed_grad_norm + p.epsilon) 
```
Is there some reason why it's different?
